src/coreclr/gc/gcpriv.h

```diff
@@ -1128,6 +1128,17 @@ enum interesting_data_point
     max_idp_count
 };
 
+enum bookkeeping_element
```

Comment:
some of these are optional depending on various defines such as `CARD_BUNDLE` and `FEATURE_USE_SOFTWARE_WRITE_WATCH_FOR_GC_HEAP`. you can just not define those at all if the define is not defined.

Reply:
This is possible, but looking at the change required to implement this feedback, I think it decreased readability quite a bit. The original code is much more compact. To show that, I intentionally made this a separate commit to make sure we understand the trade-off here.